app/code/core/Mage/ConfigurableSwatches/Helper/Productimg.php

```diff
@@ -263,6 +266,46 @@ public function getGlobalSwatchUrl(
         return Mage::getBaseUrl(Mage_Core_Model_Store::URL_TYPE_MEDIA) . $swatchImage;
     }
 
+    /**
+     * Create a swatch image for the given filename
+     *
+     * @param string $value
+     * @param int $width
+     * @param int $height
+     * @return string|null $destPath
```

Comment:
it seems to me that this method never returns null, am I missing something?

Reply:
it should return null if it fails to create the swatch from the loaded image, so it creates it from the color picker

Reply:
If I'm not reading it wrong, it doesn't seem to do that check because it returns '' on line 284 and then I don't see any return statement until line 306 but $destPath seems to always be a string

Reply:
you're right, I corrected it with false